### DISCUSSION TITLE
Apache Icebreg with Azure Blob storage?

### DISCUSSION THREAD

**Discussion by KhalithEngineer** [duckdb/duckdb#12820] [Q&A] [answered]
Hi guys, I would like to know if i could do a iceberg_scan with azure blob path, so that I can query on the iceberg table. I am using hive metastore as catalog. I could see the docs with respect to S3 storage and not on azure blob for iceberg extension.

**Comment by djouallah** [comment]
you can use this as an example

https://colab.research.google.com/drive/1dxFq_CvTC90znIyOpcSCcdJbnePd3ZWI

**Reply by KhalithEngineer** [reply, marked as answer]
@djouallah tqs man! 

I tried with mssql db as well for catalog.

```
catalog = SqlCatalog(
        name=ICEBERG_CONFIG['catalog'],
        uri='mssql+pymssql://username:password@127.0.0.1:1433/iceberg',
        warehouse=f"abfss://{os.getenv('AZURE_STORAGE_ACCOUNT_NAME')}.blob.core.windows.net/warehouse",
        **{
            "adlfs.account-name": os.getenv('AZURE_STORAGE_ACCOUNT_NAME'),
            "adlfs.account-key": os.getenv('AZURE_STORAGE_ACCOUNT_KEY'),
            "py-io-impl": "pyiceberg.io.fsspec.FsspecFileIO"
        }
    )
```
